### ISSUE TITLE
unable to program tinyFPGA BX under windows 10

### ISSUE THREAD

**Issue by simonbeaudoin0935** [tinyfpga/TinyFPGA-B-Series#20]
I was able to build a bitstream. When I try to program it under windows 10 with the gui app, it keeps saying : Unable to communicate with TinyFPGA. Reconnect and reset TinyFPGA before programming.

I keep on trying to reset the board but nothing more happen.

**Comment by RRob777** [comment]
Hello,

I have the same problem.
When I want to program my Tiny FPGA BX I get this error message: "Unable to communicate with TinyFPGA. Reconnect and reset TinyFPGA before programming.".

I am also using windows 10.
Does anyone have a solution?

**Comment by simonbeaudoin0935** [comment]
No, never worked. I wanted to throw thia shit in the garbage, but ended up
writing my own programmer using an arduino (mega, but a uno would do).
Solder SPI pins to the tinyfpga flash pads. Anyway, it worked great this way

Compile que Qt application, and then flash this bitch

https://github.com/simonbeaudoin0935/TinyFPGA-Programmer

**Comment by RRob777** [comment]
Hi.
I found the solution.
I use iCEcube2 (from Lattice) to design the VHDL code and to configure the pins used (order, layout, pull-up...). iCEcube2 then provides me a binary file (.bin).
I thought to use the TinyFPGA-Programmer software but there is a problem with it (as explained above).

--> 1. The solution to program the Tiny FPGA BX is to follow carefully the instructions of this site stopping at point 3 (before downloading Atom):

TinyFPGA BX User Guide : https://tinyfpga.com/bx/guide.html

*If you encounter problems during the installation of the modules (Apio, tinyprog...) do not hesitate to type your error message (which appeared on the Windows command prompt) on google. There are often forums that have solved problems.
**I have encountered this problem: No module named pyrsistent.
Write in the Windows command prompt (cmd): pip install "pyrsistent==0.15.7"


2. Then follow the instructions on this site from point 13 and stop after doing point 16:

https://www.centennialsoftwaresolutions.com/post/install-remove-tools-for-tinyfpga-bx-programming-and-program-the-board-on-windows-7-sp1


And there, if everything went well, you have programmed your Tiny FPGA BX.
This method allows you to program your Tiny FPGA BX with the Windows command prompt (from a binare file (.bin)).
For information, I am on Windows 10.

**Comment by RRob777** [comment]
To reprogram your Tiny FPGA BX the next time, repeat step 2 after generating your binary file (.bin).
